internal/guest/runtime/hcsv2/workload_container.go

```diff
@@ -45,6 +48,37 @@ func updateSandboxMounts(sbid string, spec *oci.Spec) error {
 	return nil
 }
 
+func updateHugePageMounts(sbid string, spec *oci.Spec) error {
+	mountPrefix := "hugepages://"
+	for i, m := range spec.Mounts {
+		if strings.HasPrefix(m.Source, mountPrefix) {
+			mountsDir := getSandboxHugePageMountsDir(sbid)
+			subPath := strings.TrimPrefix(m.Source, mountPrefix)
+			hugePageMountSource := filepath.Join(mountsDir, subPath)
+
+			// filepath.Join cleans the resulting path before returning so it would resolve the relative path if one was given.
+			// Hence, we need to ensure that the resolved path is still under the correct directory
+			if !strings.HasPrefix(hugePageMountSource, mountsDir) {
+				return errors.Errorf("mount path %v for mount %v is not within hugepages's mounts dir", hugePageMountSource, m.Source)
+			}
+
+			spec.Mounts[i].Source = hugePageMountSource
+
+			_, err := os.Stat(hugePageMountSource)
+			if os.IsNotExist(err) {
+				if err := os.MkdirAll(hugePageMountSource, 0755); err != nil {
+					return err
+				}
+
+				if err := unix.Mount("none", hugePageMountSource, "hugetlbfs", 0, "pagesize="+subPath); err != nil {
```

Comment:
we're using `subPath` still as the pagesize here but that would be the whole subpath under the hugepages mount according to the earlier code in the function, not just the size. (iow it would be 2mb/a/b/whatever instead of just 2mb)

Reply:
right, I guess, invalid value just gets dropped saved me :-).

Reply:
fixed it, will push it after addressing all the comments.